File: frontend/codeauditor/src/app/domain/models/learning-profile.ts
// LearningProfile — per-user persistent learning preferences consumed by the
// socratic tutor chat.
//
// Matches the Go struct in backend/internal/core/domain/models/learning_profile.go.
// All JSON keys are camelCase (Go json tags already use camelCase).
// Zero framework imports. Pure TypeScript domain model.
export interface Preferencias {
  idioma: string; // es | en | fr | de
  nivelSocratismo: number; // 0-3
  longitudMaximaMsg: number; // max chars per tutor message
  bloqueosTipicos: string[];
  topicsQueLeCuestan: string[];
  resumen: string;
}

export interface EstiloAprendizaje {
  aprendeMejorCon: string; // ejemplos | teoria | practica | ...
}

export interface LearningProfile {
  userId: string;
  preferencias: Preferencias;
  estiloAprendizaje: EstiloAprendizaje;
  ultimaActualizacion: string; // ISO 8601
}

File: backend/internal/core/domain/models/learning_profile.go
package models

import "time"

// Preferencias holds the user-facing learning preferences managed by the
// socratic tutor chat. Stored as a JSONB column on the DB side.
type Preferencias struct {
	Idioma             string   `json:"idioma"`
	NivelSocratismo    int      `json:"nivelSocratismo"`
	LongitudMaximaMsg  int      `json:"longitudMaximaMsg"`
	BloqueosTipicos    []string `json:"bloqueosTipicos,omitempty"`
	TopicsQueLeCuestan []string `json:"topicsQueLeCuestan,omitempty"`
	Resumen            string   `json:"resumen,omitempty"`
}

// EstiloAprendizaje captures how the user learns best. Stored as a JSONB
// column on the DB side.
type EstiloAprendizaje struct {
	AprendeMejorCon string `json:"aprendeMejorCon,omitempty"`
}

// LearningProfile is a per-user persistent learning profile consumed by the
// socratic tutor. Upserted with defaults on first read.
type LearningProfile struct {
	UserID              string            `json:"userId"`
	Preferencias        Preferencias      `json:"preferencias"`
	EstiloAprendizaje   EstiloAprendizaje `json:"estiloAprendizaje"`
	UltimaActualizacion time.Time         `json:"ultimaActualizacion"`
}

// DefaultLearningProfile returns a freshly-initialized profile with the
// spec-mandated defaults (idioma=es, nivel_socratismo=2, etc.).
func DefaultLearningProfile(userID string) *LearningProfile {
	return &LearningProfile{
		UserID: userID,
		Preferencias: Preferencias{
			Idioma:             "es",
			NivelSocratismo:    2,
			LongitudMaximaMsg:  200,
			BloqueosTipicos:    []string{},
			TopicsQueLeCuestan: []string{},
			Resumen:            "",
		},
		EstiloAprendizaje: EstiloAprendizaje{
			AprendeMejorCon: "ejemplos",
		},
	}
}
